libs/core/langchain_core/messages/base.py

```diff
@@ -95,6 +95,38 @@ def merge_content(
         else:
             return_list: List[Union[str, Dict]] = [first_content]
             return return_list + second_content
+    elif isinstance(second_content, dict) and isinstance(
+        second_content.get("index"), int
+    ):
+        second_index = second_content.get("index")
+        if second_index < 0:
+            raise Exception(
+                f"Invalid index value {second_index} found when concatenating chunks"
+            )
+        if isinstance(first_content, List):
+            if second_index > len(first_content):
+                raise Exception(
+                    f"Failed to concatenate a chunk with index {second_index}",
+                    f"to a chunk that only had {len(first_content)} parts.\n"
+                    "You must concatenate message content chunks in order.",
+                )
```

Comment:
why couldn't second chunk refer to an item after all the items in first chunk?

Reply:
It can, but it must be the very next one. Otherwise you get holey arrays, which sounded ugly. We *could* do it

Reply:
doesn't feel like something we should be enforcing imo